mypy/checker.py

```diff
@@ -3483,21 +3484,34 @@ def find_isinstance_check(self, node: Expression
                     vartype = type_map[expr]
                     return self.conditional_callable_type_map(expr, vartype)
         elif isinstance(node, ComparisonExpr):
-            # Check for `x is None` and `x is not None`.
+            operand_types = [coerce_to_literal(type_map[expr])
+                             for expr in node.operands if expr in type_map]
+
             is_not = node.operators == ['is not']
-            if any(is_literal_none(n) for n in node.operands) and (
-                    is_not or node.operators == ['is']):
+            if (is_not or node.operators == ['is']) and len(operand_types) == len(node.operands):
```

Comment:
How hard would be it be to do exactly the same for `==`? (Mostly so that example in https://github.com/python/mypy/issues/4223 will not give the false positive.)

Reply:
It's slightly trickier -- the semantics of `a is SomeEnum.Foo` and `a == SomeEnum.Foo` are different, unfortunately.

If `a` is something like an int or some other unrelated type, we know the first expression will always be False. But for the second, we have no idea since `a` could have defined a custom `__eq__` function. SomeEnum itself could also have defined/inherited a custom `__eq__` method, which would further complicate things.

I'll submit a separate PR for this: it ended up being easier to make this change if I also added support for chained operator comparisons at the same time (see below).